src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -666,10 +697,101 @@ MetadataCreateCommands(void)
 												  shardCreateCommandList);
 	}
 
+	/* As the last step, propagate the pg_dist_object entities */
+	List *distributedObjectSyncCommandList = DistributedObjectMetadataSyncCommandList();
+	metadataSnapshotCommandList = list_concat(metadataSnapshotCommandList,
+											  distributedObjectSyncCommandList);
+
 	return metadataSnapshotCommandList;
 }
 
 
+/*
+ * DistributedObjectMetadataSyncCommandList returns the necessary commands to create
+ * pg_dist_object entries on the new node.
+ */
+static List *
+DistributedObjectMetadataSyncCommandList(void)
+{
+	HeapTuple pgDistObjectTup = NULL;
+	Relation pgDistObjectRel = table_open(DistObjectRelationId(), AccessShareLock);
+	Relation pgDistObjectIndexRel = index_open(DistObjectPrimaryKeyIndexId(),
+											   AccessShareLock);
+	TupleDesc pgDistObjectDesc = RelationGetDescr(pgDistObjectRel);
+
+	List *objectAddressList = NIL;
+	List *distributionArgumentIndexList = NIL;
+	List *colocationIdList = NIL;
+
+	/* For consistent test results use ordered scan */
+	SysScanDesc pgDistObjectScan = systable_beginscan_ordered(pgDistObjectRel,
+															  pgDistObjectIndexRel, NULL,
+															  0, NULL);
+	while (HeapTupleIsValid(pgDistObjectTup = systable_getnext_ordered(pgDistObjectScan,
+																	   ForwardScanDirection)))
+	{
+		Form_pg_dist_object pg_dist_object = (Form_pg_dist_object) GETSTRUCT(
+			pgDistObjectTup);
+
+		ObjectAddress *address = palloc(sizeof(ObjectAddress));
+		int32 *colocationIdPtr = palloc(sizeof(int32));
```

Comment:
Sorry for brining up this late and again, but I think using a single integer pointer is very unnecessary here (and other calls as well). 

We should use `InvalidColocationId` and `InvalidArgIndex` values to identify these. And, inside `MarkObjectsDistributedCreateCommand` we should change `distributionArgumentIndex == NULL` to `distributionArgumentIndex == InvaludDistributionArgument` etc.

Even if there is no difference how the code executes, it is confusing to see a single integer allocated like this. 

Neither Citus or PG code-base has any examples of it, and I'd rather stick to the convention and avoid the usage. That mostly avoid any confusion (such as my-self keep realizing the unusual pattern every time I go over the code:)) 

These types of allocations are natural/expected when multiple elements are allocated.